src/Mod/Path/PathScripts/PathPocketShape.py

```diff
@@ -204,104 +235,339 @@ def pocketInvertExtraOffset(self):
+                            print(sub + "is vertical after rotation.")
                         else:
                             PathLog.error(translate('PathPocket', 'Pocket does not support shape %s.%s') % (base.Label, sub))
-
+                        
+                        if reset == True:
+                            base.Placement.Rotation = startRotation
+                            base.recompute()
+                            reset = False
+                    # End IF
+                # End FOR
+                base.Placement = resetPlacement
+                base.recompute()
+            # End FOR
+            
+            ''' This section prepared for sorting of vertical axis_angle groups and analysis for PathGeom.combineConnectedShapes()
+            # Separate elements, regroup by orientation (axis_angle combination)
+            vTags = ['X34.2']
+            vGrps = [[(2.3, 3.4, 'X')]]
+            for tup in vertTuples:
+                (face, sub, angle, axis, tag, strDep, finDep, trans) = tup
+                if tag in vTags:
+                    # Determine index of found string
+                    i = 0
+                    for orn in vTags:
+                        if orn == tag:
+                            break
+                        i += 1
+                    vGrps[i].append(tup)
+                else:
+                    vTags.append(tag)  # add orientation entry
+                    vGrps.append([tup])  # add orientation entry
+            # Remove temp elements
+            vTags.pop(0)
+            vGrps.pop(0)
+            
+            # check all faces in each axis_angle group
+            shpList = []
+            for o in range(0, len(vTags)):
+                shpList = []
+                for (face, sub, angle, axis, tag, strDep, finDep, trans) in vGrps[o]:
+                    shpList.append(face)
+                # check all faces and see if they are touching/overlapping and combine those into a compound
+                # Original Code in For loop
+                self.vertical = PathGeom.combineConnectedShapes(shpList)
+                self.vWires = [TechDraw.findShapeOutline(shape, 1, FreeCAD.Vector(0, 0, 1)) for shape in self.vertical]
+                for wire in self.vWires:
+                    w = PathGeom.removeDuplicateEdges(wire)
+                    face = Part.Face(w)
+                    face.tessellate(0.05)
+                    if PathGeom.isRoughly(face.Area, 0):
+                        PathLog.error(translate('PathPocket', 'Vertical faces do not form a loop - ignoring'))
+                    else:
+                        # self.horiz.append(face)
+                        strDep = base.Shape.BoundBox.ZMax + self.leadIn
+                        finDep = judgeFinalDepth(obj, face.BoundBox.ZMin)
+                        tup = face, sub, angle, axis, tag, strDep, finDep, trans
+                        horizTuples.append(tup)
+            # Eol
+            '''
```

Comment:
Personally I'm not very excited about commented out blocks of code (also applicable to other places) - can refactor this into it's own function and not call that, I think that would be much preferable.

Reply:
I followed your suggestion, moving the code to a local function.  I then commented out the two lines calling the new local function.  In doing this, I realized a potential vertical faces grouping scenario that may now be unavailable to users.  I may need to hold this PR, and let you submit your extension changes, then rebase the 4th-axis implementation to the updated PathPocketShape.py file.  The scenario I am thinking of would only apply when "Use Rotation" is active, not "Off."  Your thoughts?

Reply:
Not sure I understand the implications. Does it break existing functionality or does it represent a hole in the new feature? I would only hold the PR if it breaks existing functionality.

Reply:
I think it will only be a hole in the new feature. I don't think existing features will be affected with "Use Rotation" set to default of "Off."  Set to "Off" I believe existing functionality is preserved.